xarray/core/dataset.py

```diff
@@ -1007,6 +1007,34 @@ def to_netcdf(self, path=None, mode='w', format=None, group=None,
                          engine=engine, encoding=encoding,
                          unlimited_dims=unlimited_dims)
 
+    def to_zarr(self, store=None, mode='a', synchronizer=None, group=None,
+                encoding=None):
+        """Write dataset contents to a zarr group.
+
+        Parameters
+        ----------
+        store : MutableMapping or str, optional
+            Store or path to directory in file system.
+        mode : {‘r’, ‘r+’, ‘a’, ‘w’, ‘w-‘}
+            Persistence mode: ‘r’ means read only (must exist); ‘r+’ means
+            read/write (must exist); ‘a’ means read/write (create if doesn’t
+            exist); ‘w’ means create (overwrite if exists); ‘w-‘ means create
+            (fail if exists).
```

Comment:
These are all the possible `mode` options for [`zarr.open_group`](http://zarr.readthedocs.io/en/latest/api/hierarchy.html#zarr.hierarchy.open_group).
But it really only makes sense to accept the writeable options.

Reply:
thoughts on:

```Python
if mode not in [‘r+’, ‘a’, ‘w’, ‘w-‘]:
    raise ValueError('invalid mode: %s ' % mode)
```
?